src/test/compile-fail/issue-43988.rs

```diff
@@ -32,5 +31,12 @@ fn main() {
 
     #[repr]
     let _y = "123";
-    //~^^ ERROR attribute should not be applied to statements
+    //~^^ ERROR attribute should not be applied a statement
+
+
+    fn foo() {}
+
+    #[inline(ABC)]
+    foo();
+    //~^^ ERROR attribute should be applied to function
```

Comment:
All the tests are for statements, could you add a test for an expression too?
Something like `let x = #[repr] y;`

Reply:
I added this example. I needed to add a feature in the tests to get it to actually run.

Otherwise `error[E0658]: attributes on non-item statements and expressions are experimental. (see issue #15701)` stopped all of the tests early.